Fix invalid code link

kevincliao opened the pull request:
<!--
There's no need to add anything here, but feel free to add a personal message.
Please describe the changes in this PR in the commit message(s) instead, with
each commit representing one logical change. Address code review comments by
rewriting the branch rather than adding commits on top. Use force-push when
pushing the updated branch (`jj git push` does that automatically when you
rewrite a branch). Merge the PR at will once it's been approved. See
https://github.com/martinvonz/jj/blob/main/docs/contributing.md for details.
-->

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (src/config-schema.json)
- [ ] I have added tests to cover my changes

martinvonz reviewed (approved):
> Fix invalid README hyperlink

nit: this is not in the README

kevincliao commented:
> > Fix invalid README hyperlink
> 
> nit: this is not in the README

Updated 

> > Fix invalid README hyperlink
> 
> nit: this is not in the README

Updated commit title and PR title.
